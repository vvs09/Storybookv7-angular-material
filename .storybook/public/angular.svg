<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
