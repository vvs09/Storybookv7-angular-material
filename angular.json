{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}