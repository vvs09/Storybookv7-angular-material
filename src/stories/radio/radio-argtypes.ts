export const radioArgTypes = {
  color: {
    control: 'radio',
    options: ['accent', 'primary', 'warn'],
    description: 'Theme color palette for the component.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  disabled: {
    control: 'boolean',
    description:
      'Whether the radio is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  ariaDescribedby: {
    control: 'text',
    description:
      'The aria-describedby attribute is read after the elements label and field type.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  ariaLabel: {
    control: 'text',
    description: 'ARIA label for the content of the chip.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  disableRipple: {
    control: 'boolean',
    description: 'Whether ripples are disabled.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  ariaLabelledby: {
    control: 'text',
    description:
      'Users can specify the aria-labelledby attribute which will be forwarded to the input element',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  id: {
    control: 'text',
    description:
      'A unique id for the radio input. If none is supplied, it will be auto-generated.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  checked: {
    control: 'boolean',
    description: 'Whether the radio is checked.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  labelPosition: {
    control: 'radio',
    options: ['before', 'after'],
    description:
      'Whether the label should appear after or before the radio. Defaults to after',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'after' },
      type: {
        summary: 'before | after',
      },
    },
  },
  name: {
    control: 'text',
    description: 'Name value will be applied to the input element if present',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  required: {
    control: 'boolean',
    description: 'Whether the radio is required.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'boolean',
      },
    },
  },
  change: {
    control: 'object',
    description: 'Event emitted when the radios checked value changes.',
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'EventEmitter<MatRadioChange>',
      },
    },
  },
  inputId: {
    control: 'text',
    description: 'Returns the unique id for the visual hidden input.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'string',
      },
    },
  },
  value: {
    control: 'text',
    description: 'The value attribute of the native input element.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'any',
      },
    },
  },
};
