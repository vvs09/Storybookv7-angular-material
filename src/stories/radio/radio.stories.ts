import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatRadioModule, MatRadioButton } from '@angular/material/radio';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { radioArgTypes } from './radio-argtypes';

const meta: Meta<MatRadioButton> = {
  title: 'Components/Radio button',
  component: MatRadioButton,
  tags: ['autodocs'],
  argTypes: {
    ...radioArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [ReactiveFormsModule, FormsModule, MatRadioModule],
    }),
  ],
  render: (args: MatRadioButton) => ({
    props: args,
    template: `<mat-radio-button ${argsToTemplate(
      args
    )}>Option 1</mat-radio-button>`,
  }),
};

export default meta;
type Story = StoryObj<MatRadioButton>;

export const Default: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDescribedby: 'radio',
    ariaLabel: 'radio',
    ariaLabelledby: 'radio',
    disableRipple: false,
    id: '1',
    checked: false,
    labelPosition: 'after',
    name: 'label radio',
    required: false,
    inputId: '1',
    value: 'radio1',
  },
};

export const Primary: Story = {
  args: {
    ...Default.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...Default.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...Default.args,
    disabled: true,
  },
};

export const Checked: Story = {
  args: {
    ...Default.args,
    checked: true,
  },
};

export const LabelPosition: Story = {
  args: {
    ...Default.args,
    labelPosition: 'before',
  },
};

export const Required: Story = {
  args: {
    ...Default.args,
    required: true,
  },
};
