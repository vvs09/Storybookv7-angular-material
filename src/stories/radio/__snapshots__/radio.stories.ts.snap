// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Radio button Checked smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-accent mat-mdc-radio-checked"
                    ng-reflect-color="accent"
                    ng-reflect-disabled="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="true"
                    ng-reflect-label-position="after"
                    ng-reflect-name="label radio"
                    ng-reflect-required="false"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field"
         ng-reflect-label-position="after"
    >
      <div class="mdc-radio">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;

exports[`Components/Radio button Default smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-accent"
                    ng-reflect-color="accent"
                    ng-reflect-disabled="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="false"
                    ng-reflect-label-position="after"
                    ng-reflect-name="label radio"
                    ng-reflect-required="false"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field"
         ng-reflect-label-position="after"
    >
      <div class="mdc-radio">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;

exports[`Components/Radio button Disabled smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-accent"
                    ng-reflect-color="accent"
                    ng-reflect-disabled="true"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="false"
                    ng-reflect-label-position="after"
                    ng-reflect-name="label radio"
                    ng-reflect-required="false"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field"
         ng-reflect-label-position="after"
    >
      <div class="mdc-radio mdc-radio--disabled">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               disabled
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="true"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;

exports[`Components/Radio button LabelPosition smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-accent"
                    ng-reflect-color="accent"
                    ng-reflect-disabled="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="false"
                    ng-reflect-label-position="before"
                    ng-reflect-name="label radio"
                    ng-reflect-required="false"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field mdc-form-field--align-end"
         ng-reflect-label-position="before"
    >
      <div class="mdc-radio">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;

exports[`Components/Radio button Primary smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-primary"
                    ng-reflect-color="primary"
                    ng-reflect-disabled="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="false"
                    ng-reflect-label-position="after"
                    ng-reflect-name="label radio"
                    ng-reflect-required="false"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field"
         ng-reflect-label-position="after"
    >
      <div class="mdc-radio">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;

exports[`Components/Radio button Required smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-accent"
                    ng-reflect-color="accent"
                    ng-reflect-disabled="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="false"
                    ng-reflect-label-position="after"
                    ng-reflect-name="label radio"
                    ng-reflect-required="true"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field"
         ng-reflect-label-position="after"
    >
      <div class="mdc-radio">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               required
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;

exports[`Components/Radio button Warn smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <mat-radio-button class="mat-mdc-radio-button mat-warn"
                    ng-reflect-color="warn"
                    ng-reflect-disabled="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-id="1"
                    ng-reflect-checked="false"
                    ng-reflect-label-position="after"
                    ng-reflect-name="label radio"
                    ng-reflect-required="false"
                    ng-reflect-value="radio1"
                    id="1"
  >
    <div mat-internal-form-field
         class="mdc-form-field mat-internal-form-field"
         ng-reflect-label-position="after"
    >
      <div class="mdc-radio">
        <div class="mat-mdc-radio-touch-target">
        </div>
        <input type="radio"
               class="mdc-radio__native-control"
               id="1-input"
               name="label radio"
               value="radio1"
               tabindex="0"
        >
        <div class="mdc-radio__background">
          <div class="mdc-radio__outer-circle">
          </div>
          <div class="mdc-radio__inner-circle">
          </div>
        </div>
        <div mat-ripple
             class="mat-ripple mat-radio-ripple mat-mdc-focus-indicator"
             ng-reflect-trigger="[object HTMLDivElement]"
             ng-reflect-disabled="false"
             ng-reflect-centered="true"
        >
          <div class="mat-ripple-element mat-radio-persistent-ripple">
          </div>
        </div>
      </div>
      <label class="mdc-label"
             for="1-input"
      >
        Option 1
      </label>
    </div>
  </mat-radio-button>
</storybook-root>
`;
